src/main/java/use_case/podcast/PodcastInteractor.java

```diff
@@ -0,0 +1,28 @@
+package use_case.podcast;
+
+
+import data_access.PodcastDataAccess;
+import entities.Podcast;
+import java.util.UUID;
+
+public class PodcastInteractor implements PodcastInputBoundary {
+
+    private final PodcastOutputBoundary outputBoundary;
+    private final PodcastDataAccess podcastDAO;
+
+    public PodcastInteractor(PodcastOutputBoundary outputBoundary, PodcastDataAccess podcastDAO) {
+        this.outputBoundary = outputBoundary;
+        this.podcastDAO = podcastDAO;
+    }
+
+    @Override
+    public void execute(PodcastInputData podcastInputData) {
+        UUID uniqueID = UUID.randomUUID();
+        Podcast podcast = new Podcast(uniqueID, podcastInputData.getTitle(), podcastInputData.getAssignedTo(), podcastInputData.getEpisodes());
+        if (podcastDAO.savePodcast(podcast)){
+            outputBoundary.prepareSuccessView(new PodcastOutputData(podcast));
+        } else {
+            outputBoundary.prepareFailView("Failed to save podcast.");
```

Comment:
If there is only one possible error message when we call prepareFailView, maybe we don't necessarily need the error parameter.

Reply:
I believe the general design idea behind having an error parameter is if there is an additional error we want to address in the future, the program is equipped to do so.